nixos/doc/manual/development/writing-nixos-tests.section.md

```diff
@@ -34,7 +37,64 @@ when switching between consoles, and so on. An interesting multi-node test is
 [`nfs/simple.nix`](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/nfs/simple.nix).
 It uses two client nodes to test correct locking across server crashes.
 
-There are a few special NixOS configuration options for test VMs:
+## Calling a test {#sec-calling-nixos-tests}
+
+Tests are invoked a bit differently depending on whether the test lives in NixOS or in another project.
+
+### Testing within NixOS {#sec-call-nixos-test-in-nixos}
+
+Test modules can be instantiated into derivations in multiple ways.
+
+Tests that are part of NixOS are added to [`nixos/tests/all-tests.nix`](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/all-tests.nix).
+
+```nix
+  hostname = runTest ./hostname.nix;
+```
+
+Overrides can be added by defining an anonymous module in `all-tests.nix`.
+For the purpose of constructing a test matrix, [use the `matrix` options](#sec-nixos-test-matrix) instead.
+
+```nix
+  hostname = runTest { imports = [ ./hostname.nix ]; defaults.networking.firewall.enable = false; };
+```
+
+You can run a test with attribute name `mytest` in `nixos/tests/all-tests.nix` by invoking:
+
+```shell
+nix-build -A nixosTests.mytest
```

Comment:
As a beginner, one sometimes misses the intricacies, that are very obvious to veteran maintainers. I might probably execute the given command to run tests at first in `nixpkgs/nixos/tests` (because I just edited `all-tests.nix` there) or `nixpkgs/nixos/` (because I added a test to NixOS), being befuddled by the error messages, before running it in `nixpkgs/` itself, where it actually works.

```suggestion
cd /my/git/clone/of/nixpkgs
nix-build -A nixosTests.mytest
```

Reply:
:+1: , thanks!